src/plugins/intel_gna/optimizer/gna_pass_manager.cpp

```diff
@@ -978,6 +979,33 @@ void InsertCopyLayerPass::run() {
             }
         }
     }
+
+    for (auto & l : *pLayers) {
```

Comment:
From my point of view it's easier to go from the model outputs upper than from model inputs down: there is no need to check all branches and CNNNetDFS can be used to find functional layers upper the specified one. If no functional layers (excluding Splits, Concats and Crops) are found we can assume that it's an output of non-functional subgraph and Copy should be inserted.

Reply:
Thank you, done